### ISSUE TITLE
collection: separating char for setid.paramid

### ISSUE THREAD

**Issue by berndbischl** [mlr-org/paradox#212]
i am leaning now towards using
setid...paramid

so 3 dots. still valid name, but easy to split
and shouldnt occur in any name

@mb706 @mllg

**Comment by mb706** [comment]
I don't think it should be necessary to have a unique separator between setid and paramid, there should always be a way to get this information conveniently without relying on in-band information. How about a `$param_path` slot (or using the `ids()` function with another argument) that returns `map(self$params, "id")` (i.e. named list `id -> id`) in a `ParamSet` and 
```r
param_path = unlist(map(private$.sets, function(ps)
  map(ps$param_path, function(pp)
    c(ps$set_id, pp))),
  recursive = FALSE)
names(param_path) = names(self$params)
```
in a `ParamSetCollection`. I.e. it is `list(param_a = "param_a", param_b = "param_b")` for a `ParamSet` of two parameters, and `list(set1.param_a = c("set1", "param_a"), set1.param_b = c("set1", "param_b"), set2.another_param = c("set2", "another_param"))` for a `ParamSetCollection` of two `ParamSets` with two and one parameter.

**Comment by berndbischl** [comment]
we agreed that the collection should store a "reverse lookup" which list all combos of setids with params

**Comment by mb706** [comment]
Was actually fixed by #223
